src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -181,7 +177,7 @@ protected virtual HashAlgorithmName GetHashAlgorithmName(string algorithm)
             throw LogHelper.LogException<ArgumentOutOfRangeException>(LogMessages.IDX10640, algorithm);
         }
 
-        private void ResolveDotNetCoreAsymmetricAlgorithm(AsymmetricSecurityKey key, string algorithm, bool willCreateSignatures)
+        public virtual void ResolveAsymmetricAlgorithm(AsymmetricSecurityKey key, string algorithm, bool willCreateSignatures)
```

Comment:
Guys, how about the method returns
1. an AsymmetricAlgorithm, this would result in the need to cast to something we understand such as RSA or ECDsa.
2. Invent an interface similar to Deformater / Formater that can sign or verify